src/oogeso/core/simulator.py

```diff
@@ -112,36 +140,65 @@ def runSimulation(
                 solver=solver, write_yaml=write_yaml, timelimit=timelimit
             )
             # 3. Save results (for later analysis)
-            new_results = self._saveOptimisationResult(step, store_duals)
+            new_results = self._saveOptimisationResult(
+                step, return_variables, store_duals
+            )
             result_object.append_results(new_results)
             first = False
 
         return result_object
 
-    def _saveOptimisationResult(self, timestep, store_duals=None):
-        """save results of optimisation for later analysis"""
+    def _saveOptimisationResult(
+        self, timestep, return_variables, store_duals=None
+    ) -> SimulationResult:
+        """extract results of optimisation for later analysis"""
 
         # TODO: Implement result storage
         # hdf5? https://stackoverflow.com/questions/47072859/how-to-append-data-to-one-specific-dataset-in-a-hdf5-file-with-h5py)
         pyomo_instance = self.optimiser.pyomo_instance
         timelimit = self.optimiser.optimisation_parameters.optimisation_timesteps
-        data_to_keep = self.optimiser.optimisation_parameters.optimisation_return_data
-        timeshift = timestep
 
-        # TODO: Implement possibility to store subset of the data
-        if data_to_keep is not None:
-            logger.warning("Storing only a subset of the data not implemented yet.")
+        all_available_variables = [
```

Comment:
A more generic and simpler way to do this is to instead do for each if-statement do

Set return_variables as Optional

return_all = False
if not return_variables:
   return_all = True

<code here>

if return_all or "dfDeviceFlow" in return_variables:
    <code here>
if return_all or dfDeviceIsPrep in retur_variables:
    <code here>
    
and so on...

I do not see a need to change this now, but next time you make a change to this then that would make it easier in the future to maintain and update.